tests/unit/drivers/test_segmenter_driver.py

```diff
@@ -24,56 +26,60 @@ def craft(self, text: str, *args, **kwargs) -> List[Dict]:
-    def test_segment_driver(self):
-        docs = create_documents_to_segment()
-        driver = SimpleSegmentDriver(first_chunk_id=3)
-        executor = MockSegmenter()
-        driver.attach(executor=executor, pea=None)
-        driver._apply(docs[0])
-
-        assert docs[0].length == 2
-        assert docs[1].length == 2
-
-        assert docs[0].chunks[0].id == 3
-        assert docs[0].chunks[0].parent_id == docs[0].id
-        assert docs[0].chunks[0].blob == array2pb(np.array([0.0, 0.0, 0.0]))
-        assert docs[0].chunks[0].weight == 0
-        assert docs[0].chunks[0].length == 3
-
-        assert docs[0].chunks[1].id == 4
-        assert docs[0].chunks[1].parent_id == docs[0].id
-        assert docs[0].chunks[1].blob == array2pb(np.array([1.0, 1.0, 1.0]))
-        assert docs[0].chunks[1].weight == 1
-        assert docs[0].chunks[1].length == 3
-
-        assert docs[0].chunks[2].id == 5
-        assert docs[0].chunks[2].parent_id == docs[0].id
-        assert docs[0].chunks[2].blob == array2pb(np.array([2.0, 2.0, 2.0]))
-        assert docs[0].chunks[2].weight == 2
-        assert docs[0].chunks[2].length == 3
-
-        with self.assertRaises(AttributeError) as error:
-            driver._apply(docs[1])
-        assert error.exception.__str__() == '\'Document\' object has no attribute \'non_existing_key\''
+def test_segment_driver():
+    valid_doc = jina_pb2.Document()
+    valid_doc.id = 1
+    valid_doc.text = 'valid'
+    valid_doc.length = 2
+
+    driver = SimpleSegmentDriver(first_chunk_id=3)
+    executor = MockSegmenter()
+    driver.attach(executor=executor, pea=None)
+    driver._apply(valid_doc)
+
+    assert valid_doc.length == 2
+
+    assert valid_doc.chunks[0].id == 3
+    assert valid_doc.chunks[0].parent_id == valid_doc.id
+    assert valid_doc.chunks[0].blob == array2pb(np.array([0.0, 0.0, 0.0]))
+    assert valid_doc.chunks[0].weight == 0
+    assert valid_doc.chunks[0].length == 3
+    assert valid_doc.chunks[0].mime_type == "text/plain"
+
+    assert valid_doc.chunks[1].id == 4
+    assert valid_doc.chunks[1].parent_id == valid_doc.id
+    assert valid_doc.chunks[1].blob == array2pb(np.array([1.0, 1.0, 1.0]))
+    assert valid_doc.chunks[1].weight == 1
+    assert valid_doc.chunks[1].length == 3
+    assert valid_doc.chunks[1].mime_type == ""
+
+    assert valid_doc.chunks[2].id == 5
+    assert valid_doc.chunks[2].parent_id == valid_doc.id
+    assert valid_doc.chunks[2].blob == array2pb(np.array([2.0, 2.0, 2.0]))
+    assert valid_doc.chunks[2].weight == 2
+    assert valid_doc.chunks[2].length == 3
+
+
+
+def test_broken_document():
+    driver = SimpleSegmentDriver(first_chunk_id=3)
+    executor = MockSegmenter()
+    driver.attach(executor=executor, pea=None)
+
+    invalid_doc = jina_pb2.Document()
+    invalid_doc.id = 2
+    invalid_doc.text = 'invalid'
+    invalid_doc.length = 2
+
+    assert invalid_doc.length == 2
+
+    with pytest.raises(AttributeError):
+        driver._apply(invalid_doc)
```

Comment:
what does this test? cant see any meaningful `assert`

Reply:
It checks whether and `AttributeError` is raised during running the code in the `with` block. If no exception is thrown, the test will fail.

This is the pytest way of checking, that exceptions are thrown. Pretty elegant, I'd say.

Reply:
See https://docs.pytest.org/en/reorganize-docs/new-docs/user/pytest_raises.html

Reply:
Yeah Max told me about this pytest way yesterday, I think it looks cleaner

Reply:
aha, okay, missing that